### PR TITLE
crimson/{osd,heartbeat}: allow heartbeat to have access to authorizer

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @liu-chunmei @athanatos 

this change is part of #26697

File: src/crimson/net/SocketConnection.cc [hunk 1]
```diff
@@ -758,23 +758,21 @@ SocketConnection::repeat_connect()
...
-      };
-      return socket->write_flush(std::move(bl));
-    }).then([this] {
-     // read the reply
+  h.authorizer = dispatcher.ms_get_authorizer(peer_type);
+  bufferlist bl;
+  if (h.authorizer) {
+    h.connect.authorizer_protocol = h.authorizer->protocol;
+    h.connect.authorizer_len = h.authorizer->bl.length();
+    bl.append(create_static(h.connect));
```
Comment (rzarzynski) [inline src/crimson/net/SocketConnection.cc]: Looks `h` is a part of the object, and `write_flush` already assumes `this` is alive. Should be OK.

File: src/crimson/osd/chained_dispatchers.cc [hunk 2]
```diff
@@ -38,35 +38,16 @@ ChainedDispatchers::ms_handle_remote_reset(ceph::net::ConnectionRef conn) {
...
-          } else {
-            // try next one
-            ++first;
-            return {};
-          }
-        });
-      }
-    });
-  });
+  for (auto dispatcher : dispatchers) {
```
Comment (rzarzynski) [inline src/crimson/osd/chained_dispatchers.cc]: This looks much simpler now. :-)

File: src/crimson/osd/heartbeat.cc [hunk 3]
```diff
@@ -229,7 +229,7 @@ seastar::future<> Heartbeat::ms_dispatch(ceph::net::ConnectionRef conn,
   logger().info("heartbeat: ms_dispatch {} from {}",
                 *m, m->get_source());
   switch (m->get_type()) {
-  case CEPH_MSG_PING:
+  case MSG_OSD_PING:
```
Comment (rzarzynski) [inline src/crimson/osd/heartbeat.cc]:
```
src/include/ceph_fs.h:#define CEPH_MSG_PING                   2
...
src/msg/Message.h:#define MSG_OSD_PING         70
```

From `src/messages/MOSDPing.h`:
```cpp
  MOSDPing(const uuid_d& f, epoch_t e, __u8 o, utime_t s, uint32_t min_message)
    : MessageInstance(MSG_OSD_PING, HEAD_VERSION, COMPAT_VERSION),
      fsid(f), map_epoch(e), op(o), stamp(s), min_message_size(min_message)
  { }
  MOSDPing()
    : MessageInstance(MSG_OSD_PING, HEAD_VERSION, COMPAT_VERSION), min_message_size(0)
  {}
```

So it looks fine! I just regret we don't handle those codes as `enum class`.

**Pull request merged**